nemo/collections/nlp/models/nlp_model.py

```diff
@@ -38,6 +41,36 @@ def __init__(self, cfg: DictConfig, trainer: Trainer = None):
         self.bert_model = None  # Pretrained BERT encoder
         self.set_world_size(trainer)
 
+    @rank_zero_only
+    def register_bert_model(self):
+        # check if there is an encoder, warn if not
+        if self.bert_model is None:
+            raise ValueError('Instantiate self.bert_model before registering it.')
+        else:
+            # get encoder config
+            if isinstance(self.bert_model, BertEncoder):
+                # HuggingFace Transformer Config
+                self.bert_model.config.to_json_file('nn_config.json')  # name requested by jarvis team
+                self.register_artifact('nn_config.json', 'nn_config.json')
+            elif isinstance(self.bert_model, MegatronBertEncoder):
+                config_for_json = OmegaConf.to_container(self.bert_model.config)
+                with open('nn_config.json', 'w', encoding='utf-8') as f:
+                    f.write(json.dumps(config_for_json, indent=2, sort_keys=True) + '\n')
+                self.register_artifact('nn_config.json', 'nn_config.json')
```

Comment:
This is great!

I am just not sure about the name - why `nn_config`. Can we give more meaningful name, something that will 
indicate that down there we have config params of the BERT/Megatron-based encoder?

Reply:
btw. @okuchaiev is there any way to add some additional parameters to artifacts, in particular some kind of description to a given file?

Reply:
@purn3ndu please see @tkornuta-nvidia comment. I'm also not a fan of the the name `nn_config.json` as it's not very descriptive.

Reply:
hey, I went through the document, there was inconsistency - two models were supposed to have `bert_config.json`, and QA model the same artifact was named `nn_config.json`.

Standardize all them to `bert_config.json`.

Reply:
@tkornuta-nvidia the issue with `bert_config.json` is that it may contain info of models other than bert as well, for e.g. Megatron or Albert etc. I am not a fan of `nn_config.json` either. Basically, the goal of that .json is to capture the configs of the pretrained model. So we could call it `pretrain_config.json`. Open to other suggestions.

Reply:
Then how about simply `encoder_config.json`? 

@purn3ndu let's really not waste more time on this, let's have it done, cover the whole path to JARVIS, we will polish the rough ends in the next iteration.

Generally, I think the root of cause of this problem is that we are trying to put something in "name" that would really require to be described with "richer properties". Sadly, currently NeMo Model Artifacts have only this, but maybe after official release of EFF we will upstreams some solutions to NeMo and make them richer. @okuchaiev makes sense to you?